### PR TITLE
release workflow

### PR DESCRIPTION
**Pull request by kubawach**
- adds release.yaml for automatic tag release
- adds releasing.md with full documentation
- clean-up of old, unused scripts

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+on:
+  push:
+    tags:
+      - '*'
```
[the remark this answers is not in the document]
Comment (kubawach) [reply .github/workflows/release.yaml:4]: +1

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        run: echo ::set-output name=VERSION::${GITHUB_REF#refs/tags/}
+
+      - name: Create Github Release
+        id: create_github_release
+        uses: softprops/action-gh-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          name: Release ${{ steps.get_version.outputs.VERSION }}
+          files: ./wrapper/target/otel-java-lambda-wrapper-${{ steps.get_version.outputs.VERSION }}.jar
```
[the remark this answers is not in the document]
Comment (kubawach) [reply .github/workflows/release.yaml:36]: Good idea!

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+whenever a new tag is pushed to the repository.
+
+Full release procedure is as follows:
+
+* Checkout the latest version of the `main` branch.
+* Update wrapper POM version to release (remove -SNAPSHOT)
+* Set wrapper dependency for examples to release (`examples/build.gradle`)
+* Commit / push / merge PR
+* Tag that version (eg `0.0.5`) with the [annotated tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) and `git push origin <tagname>` - runs the release workflow
+* Update wrapper POM version to next `-SNAPSHOT`
```
[the remark this answers is not in the document]
Comment (kubawach) [reply RELEASING.md:14]: Yeah, I'll take a look, thanks!

File: RELEASING.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+## Releasing a new version
+
+[This GitHub Action](.github/workflows/release.yaml) builds, and releases to both OSS Maven2 repo and Github a new version of
+the wrapper
+whenever a new tag is pushed to the repository.
```
Comment (jkwatson) [inline RELEASING.md:5]:
```suggestion
[This GitHub Action](.github/workflows/release.yaml) builds, and releases a new version of the wrapper to both Maven Central and Github whenever a new tag is pushed to the repository.
```

File: RELEASING.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+Full release procedure is as follows:
+
+* Checkout the latest version of the `main` branch.
+* Update wrapper POM version to release (remove -SNAPSHOT)
+* Set wrapper dependency for examples to release (`examples/build.gradle`)
+* Commit / push / merge PR
+* Tag that version (eg `v0.0.5` - min `v` prefix!) with the [annotated tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) and `git push origin <tagname>`
```
Comment (jkwatson) [inline RELEASING.md:13]: won't this generally be pushing to `upstream` not origin? If people are using standard remote naming?

File: RELEASING.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+* Set wrapper dependency for examples to release (`examples/build.gradle`)
+* Commit / push / merge PR
+* Tag that version (eg `v0.0.5` - min `v` prefix!) with the [annotated tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) and `git push origin <tagname>` 
+* tag will run the release workflow that needs to be *approved* by a maintainer 
+* Update wrapper POM version to next `-SNAPSHOT` 
+* Set wrapper dependency for examples to next snahpshot (`examples/build.gradle`)
+* Update main `README.MD` and `outbound-context-propagation.md` files with new version number (sans -SNAPSHOT)
+* Commit / push / merge PR
+
+After the release is completed successfully, following should happen:
```
Comment (jkwatson) [inline RELEASING.md:20]:
```suggestion
After the release is completed successfully, the following should happen:
```

File: RELEASING.md (hunk 5 above)
Comment (kubawach) [reply RELEASING.md:13]: Interesting question. As this is not very well documented I had to test with my private github account. Basically for non-members github won't allow tag push to origin, while members can clone (and push to origin which is the original repo). So IMO we can leave "origin" here.

File: RELEASING.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+[This GitHub Action](.github/workflows/release.yaml) builds, and releases a new version of the wrapper to both Maven Central and Github whenever a new tag is pushed to the repository.
+
+Full release procedure is as follows:
+
+* Checkout the latest version of the `main` branch.
+* Update wrapper POM version to release (remove -SNAPSHOT)
+* Set wrapper dependency for examples to release (`examples/build.gradle`)
+* Commit / push / merge PR
+* Tag that version (eg `v0.0.5` - min `v` prefix!) with the [annotated tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) and `git push origin <tagname>`
```
Comment (iNikem) [inline RELEASING.md:11]: I don't understand "eg `v0.0.5` - min `v` prefix!". Release tag is literally `v0.0.5`, including `v`

File: RELEASING.md (hunk 5 above)
Comment (iNikem) [reply RELEASING.md:11]: I think what John means here is that if a maintainer follows the standard remote naming, then they will push to `upstream`.

File: RELEASING.md (hunk 7 above)
Comment (kubawach) [reply RELEASING.md]: Typo - should be "mind" (without v the workflow won't run).

File: RELEASING.md (hunk 5 above)
Comment (kubawach) [reply RELEASING.md]: Yeah, but maintainer can clone (so no need for upstream, it's origin). Or do we want to keep to "fork" workflow even in such cases? I have the repo cloned on my machine.

Comment (jkwatson) [reply RELEASING.md]: Whatever we go with, we should clarify what origin is referring to here.

Comment (jkwatson) [reply RELEASING.md]: Just as my 2 cents: as a maintainer, I have my own fork, and if I need to work with the upstream, I just reference upstream, so I don't have to have two separate repositories cloned locally.

**Review by jkwatson — APPROVED** [review]
let's go ahead and get this documented. we can refine the documentation as we go. Thanks!

**Pull request merged**